src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/task/TaskCopy.java

```diff
@@ -81,6 +90,39 @@ public File getSolutionLocation(String variant) {
     return config.getSolutionDir(copyId, variant);
   }
 
+  public File getSkeletonLocation(String variant) {
+    return config.getSkeletonDir(copyId, variant);
+  }
+
+  public List<String> listSkeleton(String variant) throws TaskStorageException {
+    File sourceLocation = getSkeletonLocation(variant);
+    if (!sourceLocation.exists()) {
+      return Collections.emptyList();
+    }
+
+    try {
+      List<String> result = new LinkedList<>();
```

Comment:
ArrayList is better

Reply:
I agree; this is code I reused. Will change it.